src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -199,11 +206,103 @@ public override void OnEventWritten(string name, object? payload)
                     finally
                     {
                         activity.Stop();
+                        this.RecordDuration(activity, payload, hasError: true);
                     }
                 }
 
                 break;
         }
     }
+
+    private void RecordDuration(Activity? activity, object? payload, bool hasError = false)
+    {
+        if (SqlActivitySourceHelper.DbClientOperationDuration.Enabled == false)
+        {
+            return;
+        }
+
+        var tags = new TagList
+        {
+            { SemanticConventions.AttributeDbSystem, SqlActivitySourceHelper.MicrosoftSqlServerDatabaseSystemName },
+        };
+
+        if (activity != null && activity.IsAllDataRequested)
+        {
+            if (hasError)
+            {
+                var exceptionType = activity.Tags.SingleOrDefault(t => t.Key == SemanticConventions.AttributeExceptionType);
```

Comment:
The `exception.type` attribute is currently never set on the activity because it is a new attribute and the trace instrumentation has not yet been adapted to the new conventions.

Was your plan to update the trace instrumentation to follow the new conventions in this PR? Or were you looking to split the work up some how?